src/create_elevator_array.c

```diff
@@ -0,0 +1,45 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include"structure.h"
+#include"create_elevator_array.h"
+
+/**
+creates an array for elevators with respective to input entered by user.
+
+
+Constraints
+
+checks the input parameter-no of elevators .
+
+No of elevators should be in range(1-100).
+
+Prints error message if the no of elevators is out of range and if memory allocation fails.
+
+Returns the array of elevators.
+
+**/
+
+
+
+struct elevator * create_elevator_array(int no_of_elevators)
+{
+	if((no_of_elevators==0)||(no_of_elevators>100)||(no_of_elevators<1))//checks if the input parameter is valid.
+	{
+		printf("Input parameter -Number of elevators entered is invalid \n");
+		return 0;
+	}
+	struct elevator *ele;
+	ele=malloc (no_of_elevators * sizeof(struct elevator));//creates memory for the input-no_of_elevators.
+	if(ele==NULL)//checks if the memory allocation is success
+	{
+		printf("Memory allocation failed \n");
+		return 0;
+	}
+	else
+	{
+	
```

Comment:
You also have to initialize the elevator array including initializing the passenger array inside the structure.

Reply:
Create a temp variable of type struct and than using a for loop over the array set all values of array equal to that value. Also in for loop create allocate memory for the passenger array and initialize there values as well. Set the in_elevator bool in passengers array to false.

Reply:
Please let know what values to be assigned to array elements.In this function input parameter is just the number of elevators.

Reply:
bool	moving = 0.
bool	between_floor = 0
bool	direction_up = true
bool	direction_down = false
bool    *stop_at_floor = pointer to array of bool of length no_of_floors;
unsigned int	passenger_count = 0
unsigned int	max_passenger = 10
unsigned int	cur_floor = 0
unsigned int 	timer = 0
struct passenger *passenger_arr = pointer to array of size max passengers of type passenger.